src/client/pps.go

```diff
@@ -14,28 +14,62 @@ func NewJob(jobID string) *pps.Job {
 	return &pps.Job{ID: jobID}
 }
 
-type InputType int
+var (
+	MapStrategy = &pps.Strategy{
+		Partition:      pps.Partition_BLOCK,
+		Incrementality: true,
+	}
+
+	ReduceStrategy = &pps.Strategy{
+		Partition:      pps.Partition_FILE,
+		Incrementality: false,
+	}
+
+	StreamingReduceStrategy = &pps.Strategy{
+		Partition:      pps.Partition_FILE,
+		Incrementality: true,
+	}
+
+	GlobalStrategy = &pps.Strategy{
+		Partition:      pps.Partition_REPO,
+		Incrementality: false,
+	}
+
+	DefaultStrategy = MapStrategy
 
-const (
-	InputTypeMap InputType = iota
-	InputTypeReduce
+	StrategyAliasMap = map[string]*pps.Strategy{
+		"map":              MapStrategy,
+		"reduce":           ReduceStrategy,
+		"streaming_reduce": StreamingReduceStrategy,
+		"global":           GlobalStrategy,
+	}
 )
 
-func NewJobInput(repoName string, commitID string, inputType InputType) *pps.JobInput {
+// RepoNameReserved returns whether a given repo name is reserved for internal usage
+func RepoNameReserved(name string) bool {
+	switch name {
+	case "out", "self":
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
Could we make this a const map from string to bool? I feel like that would be a bit more readable.